lib/tasks/application.rake

```diff
@@ -121,4 +121,43 @@ namespace :application do
     Rails.logger = ActiveSupport::Logger.new($stdout)
     Rails.logger.level = 1
   end
+
+  namespace :links_valid do
+    desc "Checks the validity of links in division summary"
+    task divisions: :environment do
+      base_url = "https://theyvoteforyou.org.au"
+      md = Redcarpet::Markdown.new(Redcarpet::Render::HTML)
+
+      Division.find_each do |division|
+        wiki_motion = division.wiki_motion
+
+        if wiki_motion
+          summary_in_html = md.render(wiki_motion.description)
+          parsed_data = Nokogiri::HTML.parse(summary_in_html)
+          url_hash_table = Hash.new(0)
+
+          tags = parsed_data.xpath("//a")
+          tags.each do |tag|
+            url_from_page = tag[:href]
+            url = url_from_page
+
+            url = "http://#{url}" if !url[%r{\Ahttp://}] && !url[%r{\Ahttps://}] # adds http to the start of the URL to avoid failure
+
+            begin
+              if (url_hash_table[url]).zero?
+                url_hash_table[url] += 1
+                uri = URI(url)
+                Net::HTTP.get(uri)
+              end
+            rescue StandardError
+              puts "---------------------------------------------------------------------------------------------"
+              puts "URL seems broken! URL: #{url_from_page}"
```

Comment:
From a user point of view if I want to find out about broken links and I fix them division by division. I want to get all the broken links on a particular division in one block. Something like:
```
There are broken links in the description for division https://theyvoteforyou.org.au/divisions/senate/...:
   /foo/bar
   /anotherlink/to/somewhere else
```